crypto/merkle/proof_simple_value.go

```diff
@@ -0,0 +1,91 @@
+package merkle
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	cmn "github.com/tendermint/tendermint/libs/common"
+)
+
+const ProofOpSimpleValue = "simple:v"
+
+// SimpleValueOp takes a key and a single value as argument and
+// produces the root hash.  The corresponding tree structure is
+// the SimpleMap tree.  SimpleMap takes a Hasher, and currently
+// Tendermint uses aminoHasher.  SimpleValueOp should support
+// the hash function as used in aminoHasher.  TODO support
+// additional hash functions here as options/args to this
+// operator.
+//
+// If the produced root hash matches the expected hash, the
+// proof is good.
+type SimpleValueOp struct {
+	// Encoded in ProofOp.Key.
+	key []byte
+
+	// To encode in ProofOp.Data
+	Proof *SimpleProof `json:"simple-proof"`
+}
+
+var _ ProofOperator = SimpleValueOp{}
+
+func NewSimpleValueOp(key []byte, proof *SimpleProof) SimpleValueOp {
+	return SimpleValueOp{
+		key:   key,
+		Proof: proof,
+	}
+}
+
+func SimpleValueOpDecoder(pop ProofOp) (ProofOperator, error) {
+	if pop.Type != ProofOpSimpleValue {
+		return nil, cmn.NewError("unexpected ProofOp.Type; got %v, want %v", pop.Type, ProofOpSimpleValue)
+	}
+	var op SimpleValueOp // a bit strange as we'll discard this, but it works.
```

Comment:
Why not just marshal the proof directly rather than the op struct wrapper? ie.

```
func (op SimpleValueOp) ProofOp() ProofOp {
	bz := cdc.MustMarshalBinary(op.Proof)
         ...
```